src/gasPriceOracle/oracle.ts

```diff
@@ -0,0 +1,92 @@
+import { BigNumber, providers } from "ethers";
+import { toBN } from "../utils";
+
+export type GasPriceEstimate = {
+  maxFeePerGas: BigNumber;
+  maxPriorityFeePerGas?: BigNumber;
+};
+
+type ValidatedFeeData = {
+  gasPrice: BigNumber;
+  maxFeePerGas: BigNumber;
+  maxPriorityFeePerGas: BigNumber;
+};
+
+interface GasPriceFeed {
+  (provider: providers.Provider): Promise<GasPriceEstimate>;
+}
+
+async function getProviderFeeData(provider: providers.Provider): Promise<ValidatedFeeData> {
+  const validateFeeData = (val: BigNumber | null): BigNumber  => {
+    return BigNumber.isBigNumber(val) && val.gt(0) ? val : toBN(Number.MAX_SAFE_INTEGER);
+  };
+
+  // Not much to do on a failed retrieval, so just suppress it.
+  const response: providers.FeeData = await provider.getFeeData().catch(() => {
+    return {} as providers.FeeData
+  });
+
+  const feeData: ValidatedFeeData = {
+    "gasPrice": validateFeeData(response["gasPrice"]),
```

Comment:
Nit: why did you quote these keys?

Reply:
Looks like a hangover from a previous iteration of producing the feeData object. Good call.